src/segments/lastfm.go

```diff
@@ -0,0 +1,159 @@
+package segments
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+)
+
+type LastFM struct {
+	props properties.Properties
+	env   platform.Environment
+
+	Artist string
+	Track  string
+	Full   string
+	URL    string
+	Icon   string
+	Status string
+}
+
+const (
+	// APIKey openweathermap api key
+	lfmAPIKey properties.Property = "apiKey"
+	// LastFM username
+	lfmUsername properties.Property = "username"
+	// Seconds since last played until the song is considered stopped
+	/* lfmStoppedSeconds properties.Property = "stopped_seconds" */ // Commented out incase we want to re-enable this feature
+	// CacheKeyResponse key used when caching the response
+	lfmCacheKeyResponse string = "lfm_response"
+	// CacheKeyURL key used when caching the url responsible for the response
+	lfmCacheKeyURL string = "lfm_url"
+
+	// PlayingIcon indicates a song is playing
+	lfmPlayingIcon properties.Property = "playing_icon"
+	// StoppedIcon indicates a song is stopped
+	lfmStoppedIcon properties.Property = "stopped_icon"
+)
+
+type lmfDate struct {
+	UnixString string `json:"uts"`
+}
+type lfmTrackInfo struct {
+	IsPlaying *string `json:"nowplaying,omitempty"`
+}
+type Artist struct {
+	Name string `json:"#text"`
+}
+type lfmTrack struct {
+	Artist `json:"artist"`
+	Name   string        `json:"name"`
+	Info   *lfmTrackInfo `json:"@attr"`
+	Date   lmfDate       `json:"date"`
+}
+type tracks struct {
+	Tracks []lfmTrack `json:"track"`
+}
+
+type lfmDataResponse struct {
+	TracksInfo tracks `json:"recenttracks"`
+}
+
+func (d *LastFM) Enabled() bool {
+	err := d.setStatus()
+
+	if err != nil {
+		d.env.Error(err)
+		return false
+	}
+
+	return true
+}
+
+func (d *LastFM) Template() string {
+	return " {{ .Icon }}{{ if ne .Status \"stopped\" }}{{ .Full }}{{ end }} "
+}
+
+func (d *LastFM) getResult() (*lfmDataResponse, error) {
+	cacheTimeout := d.props.GetInt(properties.CacheTimeout, 0)
+	response := new(lfmDataResponse)
+	if cacheTimeout > 0 {
```

Comment:
```suggestion

	if cacheTimeout > 0 {
```

Please add a space here